File: 205-isomorphic-strings/isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char,char> mp1; // this map s char-> t char
        unordered_map<char,char> mp2; //    this map t char-> t char
        for(int i =0;i<s.size();i++){
            if(mp1.count(s[i]) && mp1[s[i]]!=t[i])
                return false;
            if(mp2.count(t[i]) && mp2[t[i]]!=s[i])
                return false;    
            mp1[s[i]]=t[i];
            mp2[t[i]]=s[i];
        } 
        return true;
    }
};

File: 205-isomorphic-strings/isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int[] mapS = new int[256];
        int[] mapT = new int[256];

        for (int i = 0; i < s.length(); i++) {
            char charS = s.charAt(i);
            char charT = t.charAt(i);

            // If the last seen positions of the characters don't match, return false
            if (mapS[charS] != mapT[charT]) {
                return false;
            }

            // Store the position (i + 1 to distinguish from default value 0)
            mapS[charS] = i + 1;
            mapT[charT] = i + 1;
        }
        return true;
    }
}